returnn/hdf.py

```diff
@@ -349,3 +361,5 @@ def run(self, task_id):
             )
 
         out_hdf.close()
+
+        self.excluded_segment_list.extend(excluded_segments)
```

Comment:
Mh.. do you have a way to test this? I fear that since both tasks run on different machines in different python processes, the `self.excluded_segment_list` will be reset for the "merge" subtask.

Reply:
Yes this will unfortunately not work, the job.save is not re-written, it has to be written to a file in the work dir, so `write_paths_to_file(segments, f"excluded_segments_{task_id}")` or so

Reply:
I ran the job and was successful. But now I am not as sure as before, since your argument might also be right.

Reply:
> If it was somehow successful

@JackTemaki @michelwi I was wrong, that was the old `hash`, and the new code did not work out.